tests/tst_ncio_mpi.F90

```diff
@@ -0,0 +1,196 @@
+! This is a test for the NCEPLIBS-ncio package.
+! Author: Brian Curtis (brian.curtis@noaa.gov)
+program tst_ncio_mpi
+
+  use mpi
+  use netcdf
+  use module_ncio
+  implicit none
+
+  character(len=72) charatt, time_units
+  type(Dataset) :: dset, dsetin
+  type(Variable) :: var
+  real(4), allocatable, dimension(:) :: values_1d
+  real(4), allocatable, dimension(:,:) :: values_2d
+  real(4), allocatable, dimension(:,:,:) :: values_3d
+  real(4), allocatable, dimension(:,:,:,:) :: values_4d
+  real(4), allocatable, dimension(:,:,:,:,:) :: values_5d
+  real(4), dimension(10,10) :: quantize1, quantize2
+  real(4) mval,r4val,qerr
+  integer ndim,nvar,ndims,ival,idate(6),icheck(6),ierr,n,nbits
+  integer :: my_rank, nprocs
+  integer :: mpi_err
+  logical hasit
+
+  call mpi_init(mpi_err)
+  call MPI_Comm_rank(MPI_COMM_WORLD, my_rank, mpi_err)
+  call MPI_Comm_size(MPI_COMM_WORLD, nprocs, mpi_err)
+
+  if (my_rank .eq. 0) print*, '*** Testing NCEPLIBS-ncio with MPI.'
+  dsetin = open_dataset('dynf000_template.nc.in', paropen=.true., mpicomm=MPI_COMM_WORLD)
+
+  if (my_rank .eq. 0) print *,'*** Test creation of new dataset from template...'
+  dset = create_dataset('dynf000.nc', dsetin, paropen=.true., mpicomm=MPI_COMM_WORLD)
```

Comment:
both serial and parallel tests are reading from and writing to file of the same name. 
This means one will not be able to run the tests in parallel and this needs to change.

Reply:
I assume we can read from the template either way, and i've made the dynf000.nc into dynf000_par.nc for both create/open

Reply:
It is not safe if the tests are run in parallel and each test reads from the same file.